source/server/listener_manager_impl.cc

```diff
@@ -553,9 +561,24 @@ void ListenerManagerImpl::startWorkers(GuardDog& guard_dog) {
   }
 }
 
-void ListenerManagerImpl::stopListeners() {
-  for (const auto& worker : workers_) {
-    worker->stopListeners();
+void ListenerManagerImpl::stopListeners(StopListenersType listeners_type) {
+  listeners_type_ = listeners_type;
+  for (Network::ListenerConfig& listener : listeners()) {
+    for (const auto& worker : workers_) {
+      if (listeners_type_.has_value() &&
```

Comment:
This must have a value, right? You set it above.

Reply:
Yeah :-). Morning needs more coffee..